src/Reflection/ReflectionClass.php

```diff
@@ -490,14 +490,16 @@ public function getImmediateConstants(): array
     /**
      * Get an associative array of the defined constants in this class,
      * with keys as constant names and values as constant values.
+
+     * @see ReflectionClass::getConstants() for the usage of filter
      *
      * @return array<string, scalar|array<scalar>|null>
      */
-    public function getConstants(): array
+    public function getConstants(?int $filter = null): array
```

Comment:
Given that child classes may extend this, I think it's to be labeled as BC break :thinking:

Reply:
Thanks for pointing this out, the filter logic has been moved to the adapter.  :+1: